pages/k8s/cni-overview.md

```diff
@@ -0,0 +1,53 @@
+---
+wrapper_template: "base_docs.html"
+markdown_includes:
+  nav: "shared/_side-navigation.md"
+context:
+  title: "CNI overview"
+  description: How to manage and deploy Kubernetes with flannel, calico, canal or Tigera Secure EE
+keywords: CNI, networking
+tags: [operating]
+sidebar: k8smain-sidebar
+permalink: cni-overview.html
+layout: [base, ubuntu-com]
+toc: False
+---
+
+Managing a network where containers can interoperate efficiently is very
+important. Kubernetes has adopted the Container Network Interface(CNI)
+specification for managing network resources on a cluster. This relatively
+simple specification makes it easy for Kubernetes to interact with a wide range
+of CNI-based software solutions.
+
+With **Charmed Kubernetes**, these networking 'plug-ins' are deployed as
+subordinate charms with each  node running as a `kubernetes-master` or
+`kubernetes-worker`, and ensure the smooth running of the cluster. It is
+possible to choose one of several different CNI providers for **Charmed
+Kubernetes**, which are listed below:
+
+## Supported CNI options
+
+The currently supported CNI solutions for **Charmed Kubernetes** are:
+
+ -   [Flannel][flannel]
+ -   [Calico][calico]
+ -   [Canal][canal]
+ -   [Tigera Secure EE][tigera]
+
+By default, **Charmed Kubernetes** will deploy the cluster using flannel. To chose a different CNI provider, see the individual links above.
+
+## Migrating to a different CNI solution
```

Comment:
in two minds about whether to just leave this out, though I feel it is a legitimate thing which users would ask about

Reply:
This seems good to have. I can imagine wanting to switch from flannel to calico or something like that, and the brief statement here would save me time trying to do something I can't. Likewise, I think an observant casual reader might come across this and realize that they should consider their initial CNI choice more carefully than they otherwise would.